src/redux/movies/movies.async.actions.js

```diff
@@ -0,0 +1,68 @@
+import Api from '../../helpers/Api';
+import {
+  activeMovieDataUpdated,
+  actorsDataLoaded,
+  changeLikes,
+  changeMovieRating,
+  deleteMovie,
+  editMovie,
+  moviesLoaded,
+  moviesLoadingFailed,
+  moviesLoadingStart
+} from './movies.actions';
+
+export const addLikeAsync = (movieId, movieData) => async (dispatch) => {
+  Api.updateMovieData(movieId, movieData).then(() => dispatch(changeLikes(movieId, movieData)))
+};
+
+export const changeRatingAsync = (movieId, movieData, newRating) => async (dispatch) => {
+  Api.updateMovieData(movieId, movieData).then(() => dispatch(changeMovieRating(movieId, newRating)))
+};
+
+export const fetchUpdateMovie = (movieId, movieData) => async (dispatch) => {
+  dispatch(moviesLoadingStart());
+  try {
+    const { data } = await Api.updateMovieData(movieId, movieData);
+
+    dispatch(editMovie(movieId, data));
+    dispatch(activeMovieDataUpdated(data))
+  } catch {
+    dispatch(moviesLoadingFailed())
+  }
+};
```

Comment:
why do you use different type of promise handling ?
use, please, async-await everywhere with try-catch, it's not consistent now